etl/db_utils.py

```diff
@@ -237,6 +239,10 @@ def upsert_namespace(self, name: str, description: str) -> int:
     def upsert_source(self, name: str, description: str, dataset_id: int) -> int:
         # There is no UNIQUE key constraint we can rely on to prevent duplicates
         # so we have to do a SELECT before INSERT...
+        # TODO: changing source description and rerunning grapher inserts a new source
+        # without deleting the old one. So we end up with multiple sources in grapher DB.
+        # We could delete & create them on every dataset update, but we risk losing manual
+        # changes to those sources.
```

Comment:
Is the source information usually greyed out in the admin for things that come from the data team, locked from being edited? If so, we could do the same here

Reply:
Yes, that's fortunately what happens. I've updated the note and also created an issues for cleaning up objects like Sources in DB after rerunning grapher multiple times https://github.com/owid/etl/issues/147.